dartagnan/src/main/java/com/dat3m/dartagnan/parsers/program/visitors/VisitorLitmusRISCV.java

```diff
@@ -229,12 +230,27 @@ public Object visitBranchCond(LitmusRISCVParser.BranchCondContext ctx) {
 
 	@Override
 	public Object visitFence(LitmusRISCVParser.FenceContext ctx) {
-		return programBuilder.addChild(mainThread, EventFactory.newFence("Fence." + ctx.fenceMode().mode));
+        Event fence = switch(ctx.fenceMode().mode) {
+            case "r.r" -> EventFactory.RISCV.newRRFence();
+            case "r.w" -> EventFactory.RISCV.newRWFence();
+            case "r.rw" -> EventFactory.RISCV.newRRWFence();
+            case "w.r" -> EventFactory.RISCV.newWRFence();
+            case "w.w" -> EventFactory.RISCV.newWWFence();
+            case "w.rw" -> EventFactory.RISCV.newWRWFence();
+            case "rw.r" -> EventFactory.RISCV.newRWRFence();
+            case "rw.w" -> EventFactory.RISCV.newRWWFence();
+            case "rw.rw" -> EventFactory.RISCV.newRWRWFence();
+            case "tso" -> EventFactory.RISCV.newTsoFence();
+            case "i" -> EventFactory.RISCV.newSynchronizeFence();
+            default -> throw new ParsingException("No support fence mode");
```

Comment:
Add the wrong fence mode as part of the error message, so it is easy to see what is wrong.
I would also say "Invalid fence mode" cause "no support/unsupported" sounds like we have an incomplete implementation (do we?).

Reply:
We are complete. We even support the synchronization fence `.i` which is used in several litmus tests, but it is really not part of the cat file (I noticed this the other day). I double checked both in herd's repo and RISCV documentation, and indeed this fence is not part of the formal memory model.